apps/src/blockly/addons/cdoFieldButton.js

```diff
@@ -1,116 +1,77 @@
 import GoogleBlockly from 'blockly/core';
 
-const CORNER_RADIUS = 3;
-const INNER_HEIGHT = 16;
-
+/**
+ * This is a customized field which the user clicks to select an option from a customized picker,
+ * for example, the location of a sprite from a grid or a sound file from a customized modal.
+ * @param value The initial value of the field.
+ * @param validator A function that is called to validate changes to the field's value.
+ * Takes in a value & returns a validated value, or null to abort a change
+ * @param onChange The function that handles the field's editor.
+ * @param onDisplay The function tht handles how the field text is displayed.
```

Comment:
minor: same with this function - since it's functioning more like a text presenter/transformer rather than a callback, maybe something like transformText or displayText? I also wonder if this should be marked as optional, and we should default to just displaying the text if this is isn't defined - i'm thinking about cases in which we may want to use this button and just display the text directly as is (or wrapped in a String())?

Reply:
Yeah, I like `transformText` too! And yes, I agree that we should default to displaying the text returned by `getText()` if  `transformText` is not defined. I added a condition to do this. Great catch!